File: frontend/src/types/inference.ts
export interface DetectionBox {
  class_id: number;
  class_name: string;
  x_center: number;
  y_center: number;
  width: number;
  height: number;
  confidence: number;
}

export interface SpeedInfo {
  preprocess_ms: number;
  inference_ms: number;
  postprocess_ms: number;
  total_ms: number;
  fps: number;
}

export interface ImageInferenceResult {
  filename: string;
  boxes: DetectionBox[];
  annotated_url: string;
  speed: SpeedInfo;
}

export interface MultiImageInferenceResponse {
  results: ImageInferenceResult[];
}

export interface WebcamFrameResponse {
  boxes: DetectionBox[];
  speed: SpeedInfo;
}

export interface InferenceParams {
  run_name: string;
  weights: "best" | "last";
  device: string;
  conf: number;
  iou: number;
}

File: backend/app/schemas/inference.py
from pydantic import BaseModel


class DetectionBox(BaseModel):
    class_id: int
    class_name: str
    x_center: float
    y_center: float
    width: float
    height: float
    confidence: float


class SpeedInfo(BaseModel):
    preprocess_ms: float
    inference_ms: float
    postprocess_ms: float
    total_ms: float
    fps: float


class ImageInferenceResult(BaseModel):
    filename: str
    boxes: list[DetectionBox]
    annotated_url: str
    speed: SpeedInfo


class MultiImageInferenceResponse(BaseModel):
    results: list[ImageInferenceResult]


class WebcamFrameResponse(BaseModel):
    boxes: list[DetectionBox]
    speed: SpeedInfo
